base/inference.jl

```diff
@@ -3371,9 +3371,10 @@ function invoke_NF(argexprs, etype::ANY, atypes, sv, atype_unlimited::ANY,
             end
         end
         newvar = newvar!(sv, atypes[1])
-        stmts = Any[invoke_fexpr, :($newvar = $(argexprs[1])),
+        stmts = Any[invoke_fexpr,
+                    :($newvar = $(argexprs[2])),
                     invoke_texpr]
-        argexprs[1] = newvar
+        argexprs[2] = newvar
```

Comment:
I think this happens to be after anywhere we actually care about the type and no one have written code where the evaluation of `invoke` itself has side effect......